.github/workflows/release.yml

```diff
@@ -9,43 +9,120 @@ on:
 jobs:
   builds:
     runs-on: ${{ matrix.os }}
+    
     strategy:
       matrix:
         os: [ubuntu-latest, macos-latest, windows-latest]
+        CIBW_BUILD:
+          # - cp36-*
+          - cp37-*
+          - cp38-*
+          - cp39-*
+          - cp310-*
+          - cp311-*
+          - cp312-*
 
+        CIBW_ARCHS:
+          # autobuilds
+          - auto64
+          - auto32
+          # apple silicon
+          - arm64
+          # add windows arm64
+          - ARM64
+          # linux other archs
+          - aarch64
+          - ppc64le
+          - s390x
+
+        exclude:
+          # exclude mac os specific exclusions
+          - os: ubuntu-latest
+            CIBW_ARCHS: arm64
+          - os: windows-latest
+            CIBW_ARCHS: arm64
+          # windows specific exclusions
+          - os: ubuntu-latest
+            CIBW_ARCHS: ARM64
+          - os: macos-latest
+            CIBW_ARCHS: ARM64
+          # linux specific exclusions
+          - os: macos-latest
+            CIBW_ARCHS: aarch64
+          - os: macos-latest
+            CIBW_ARCHS: ppc64le
+          - os: macos-latest
+            CIBW_ARCHS: s390x
+          - os: windows-latest
+            CIBW_ARCHS: aarch64
+          - os: windows-latest
+            CIBW_ARCHS: ppc64le
+          - os: windows-latest
+            CIBW_ARCHS: s390x
+          # macos does not support 32-bit builds
+          - os: macos-latest
+            CIBW_ARCHS: auto32
+          # apple silicon support only came from python 3.8
+          - os: macos-latest
+            CIBW_ARCHS: arm64
+            CIBW_BUILD: cp36-*
+          - os: macos-latest
+            CIBW_ARCHS: arm64
+            CIBW_BUILD: cp37-*
+          # windows ARM64 is from version 3.9
+          - os: windows-latest
+            CIBW_ARCHS: ARM64
+            CIBW_BUILD: cp36-*
+          - os: windows-latest
+            CIBW_ARCHS: ARM64
+            CIBW_BUILD: cp37-*
+          - os: windows-latest
+            CIBW_ARCHS: ARM64
+            CIBW_BUILD: cp38-*
+
+        # specific archs excluded by auto builds
+        include:
+          # additional archs for linux
+          - os: ubuntu-latest
+            CIBW_ARCHS: aarch64
+          - os: ubuntu-latest
+            CIBW_ARCHS: ppc64le
+          - os: ubuntu-latest
+            CIBW_ARCHS: s390x
+          # additional archs for windows
+          - os: windows-latest
+            CIBW_ARCHS: ARM64
```

Comment:
IMO this seems a bit complex to manage, due to the compatibility matrix.

Shall we have separate configs for different OS?